src/panels/lovelace/editor/config-elements/hui-calendar-card-editor.ts

```diff
@@ -0,0 +1,134 @@
+import type { LovelaceCardEditor } from "../../types";
+import { processEditorEntities } from "../process-editor-entities";
+import type { EditorTarget, EntitiesEditorEvent } from "../types";
+import { entitiesConfigStruct } from "../types";
+import { configElementStyle } from "./config-elements-style";
+
+const cardConfigStruct = struct({
+  type: "string",
+  title: "string|number?",
+  theme: "string?",
+  entities: [entitiesConfigStruct],
+});
+
+@customElement("hui-calendar-card-editor")
+export class HuiCalendarCardEditor extends LitElement
+  implements LovelaceCardEditor {
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: CalendarCardConfig;
+
+  @property() private _configEntities?: EntitiesCardEntityConfig[];
+
+  public setConfig(config: CalendarCardConfig): void {
+    config = cardConfigStruct(config);
+    this._config = config;
+    this._configEntities = processEditorEntities(config.entities);
+  }
+
+  get _title(): string {
+    return this._config!.title || "";
+  }
+
+  get _theme(): string {
+    return this._config!.theme || "";
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    return html`
+      ${configElementStyle}
+      <div class="card-config">
+        <div class="side-by-side">
+          <paper-input
+            .label="${this.hass.localize(
+              "ui.panel.lovelace.editor.card.generic.title"
+            )} (${this.hass.localize(
+              "ui.panel.lovelace.editor.card.config.optional"
+            )})"
+            .value=${this._title}
+            .configValue=${"title"}
+            @value-changed=${this._valueChanged}
+          ></paper-input>
+          <hui-theme-select-editor
+            .hass=${this.hass}
+            .value=${this._theme}
+            .configValue=${"theme"}
+            @value-changed=${this._valueChanged}
+          ></hui-theme-select-editor>
+        </div>
+      </div>
+      <h3>
+        ${"Calendar Entities" +
+        " (" +
+        this.hass!.localize("ui.panel.lovelace.editor.card.config.required") +
+        ")"}
+      </h3>
+      <ha-entities-picker
+        .hass=${this.hass}
+        .value=${this._configEntities?.map((e) => e.entity)}
+        .includeDomains=${["calendar"]}
+        @value-changed=${this._valueChanged}
+      >
+      </ha-entities-picker>
+    `;
+  }
+
+  private _valueChanged(ev: EntitiesEditorEvent | CustomEvent): void {
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const target = ev.target! as EditorTarget;
+
+    if (
+      (target.configValue! === "title" && target.value === this._title) ||
+      (target.configValue! === "theme" && target.value === this._theme)
+    ) {
+      return;
+    }
+
+    if (ev.detail && ev.detail.value && Array.isArray(ev.detail.value)) {
+      this._config.entities = ev.detail.value;
+      this._configEntities = processEditorEntities(this._config.entities);
```

Comment:
`processEditorEntities` should be used when the entities should be objects, we can just use an array here with entityIds right?